tests/test_geometry_3d.h

```diff
@@ -0,0 +1,417 @@
+				ClippingPlane(clippingPlane), Polygon(polygon), WantEqual(wantEqual){};
+	};
+	Vector<Case> tt;
+	Vector<Plane> boxPlanes = Geometry3D::build_box_planes(Vector3(5, 10, 5));
+	Vector<Vector3> box = Geometry3D::compute_convex_mesh_points(&boxPlanes[0], boxPlanes.size());
+	tt.push_back(Case(Plane(), box, true));
+	tt.push_back(Case(Plane(Vector3(0, 3, 0), Vector3(0, 1, 0)), box, false));
+	for (int i = 0; i < tt.size(); ++i) {
+		Case currentCase = tt[i];
+		Vector<Vector3> output = Geometry3D::clip_polygon(currentCase.Polygon, currentCase.ClippingPlane);
+		if (currentCase.WantEqual) {
+			CHECK(output == currentCase.Polygon);
+		} else {
+			CHECK(output != currentCase.Polygon);
+		}
+	}
+}
+TEST_CASE("[Geometry3D] Compute Convex Mesh Points") {
+	struct Case {
+		Vector<Plane> Mesh;
+		Vector<Vector3> Want;
+		Case(){};
+		Case(Vector<Plane> mesh, Vector<Vector3> want) :
+				Mesh(mesh), Want(want){};
+	};
+	Vector<Case> tt;
+	Vector<Vector3> cube;
+	cube.push_back(Vector3(-5, -5, -5));
+	cube.push_back(Vector3(5, -5, -5));
+	cube.push_back(Vector3(-5, 5, -5));
+	cube.push_back(Vector3(5, 5, -5));
+	cube.push_back(Vector3(-5, -5, 5));
+	cube.push_back(Vector3(5, -5, 5));
+	cube.push_back(Vector3(-5, 5, 5));
+	cube.push_back(Vector3(5, 5, 5));
+	tt.push_back(Case(Geometry3D::build_box_planes(Vector3(5, 5, 5)), cube));
+	for (int i = 0; i < tt.size(); ++i) {
+		Case currentCase = tt[i];
+		Vector<Vector3> vectors = Geometry3D::compute_convex_mesh_points(&currentCase.Mesh[0], currentCase.Mesh.size());
+		CHECK(vectors == currentCase.Want);
+	}
+}
+TEST_CASE("[Geometry3D] Get Closest Point To Segment") {
+	struct Case {
+		Vector3 Point;
+		Vector<Vector3> Segment;
+		Vector3 Want;
+		Case(){};
+		Case(Vector3 point, Vector<Vector3> segment, Vector3 want) :
+				Point(point), Segment(segment), Want(want){};
+	};
+	Vector<Case> tt;
+	Vector<Vector3> testSegment;
+	testSegment.push_back(Vector3(1, 1, 1));
+	testSegment.push_back(Vector3(5, 5, 5));
+	tt.push_back(Case(Vector3(2, 1, 4), testSegment, Vector3(2.33333, 2.33333, 2.33333)));
+	for (int i = 0; i < tt.size(); ++i) {
+		Case currentCase = tt[i];
+		Vector3 output = Geometry3D::get_closest_point_to_segment(currentCase.Point, &currentCase.Segment[0]);
+		CHECK(output.is_equal_approx(currentCase.Want));
+	}
+}
+TEST_CASE("[Geometry3D] Plane and Box Overlap") {
+	struct Case {
+		Vector3 Normal, MaxBox;
+		float D;
+		bool Want;
+		Case(){};
+		Case(Vector3 normal, float d, Vector3 maxBox, bool want) :
+				Normal(normal), MaxBox(maxBox), D(d), Want(want){};
+	};
+	Vector<Case> tt;
+	tt.push_back(Case(Vector3(3, 4, 2), 5, Vector3(5, 5, 5), true));
+	tt.push_back(Case(Vector3(0, 1, 0), -10, Vector3(5, 5, 5), false));
+	tt.push_back(Case(Vector3(1, 0, 0), -6, Vector3(5, 5, 5), false));
+	for (int i = 0; i < tt.size(); ++i) {
+		Case currentCase = tt[i];
+		bool overlap = Geometry3D::planeBoxOverlap(currentCase.Normal, currentCase.D, currentCase.MaxBox);
```

Comment:
So I guess I understand why you've chosen the `camelCase`, because some core methods do not follow this convention...

This should be renamed to `plane_box_overlap`, not necessarily in this PR but that should be done, seems like the code is copy-pasted from Bullet source. 🙂